client-side/src/graphs/graphs.jsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import PropTypes from 'prop-types'
+import { PieChart, ChartsLegend } from '@mui/x-charts';
+import Stack from '@mui/material/Stack';
+import { useQuery } from '@apollo/client';
+import { GET_USERS, GET_WEBSITE } from './queries.js';
+import { useAppSelector } from '../redux/store.jsx'
+
+const VisitedWebsitesComponent = ({ startDate, endDate }) => {
+    const websites = useQuery(GET_WEBSITE);
+    const users = useQuery(GET_USERS);
+
+    const getRandomColor = () => `#${Math.floor(Math.random() * 16777215).toString(16)}`;
+
+    const user = useAppSelector((state) => state.user.currentUser)
+
+    function formatDate(date) {
+        let year = date.getFullYear();
+        let month = date.getMonth() + 1
+        let day = date.getDate()
+        let d = `${year}-${month}-${day}`
+        return d;
+    }
+
+    let dateStart = new Date(startDate.$d);
+    dateStart = formatDate(dateStart)
+    let dateEnd = new Date(endDate.$d);
+    dateEnd = formatDate(dateEnd)
+    if (websites.loading || users.loading) return <p>Loading...</p>;
+
+    if (websites.error || users.error) return <p>Error</p>;
+
+    function getWebsites(dateStart, dateEnd) {
```

Comment:
logic should be placed in separated files, for example graphsUtils.js, and be called from the component